core/src/main/java/greencity/config/WebMvcConfig.java

```
@@ -96,6 +106,37 @@ public void addInterceptors(InterceptorRegistry registry) {
 
     @Override
     public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
+        resolvers.removeIf(resolver -> resolver instanceof PageableHandlerMethodArgumentResolver);
         resolvers.add(new UserArgumentResolver(userService, modelMapper));
+        resolvers.add(new PageableHandlerMethodArgumentResolver() {
+            @Override
+            public Pageable resolveArgument(MethodParameter methodParameter,
+                ModelAndViewContainer mavContainer,
+                NativeWebRequest webRequest,
+                WebDataBinderFactory binderFactory) {
+                String pageParam = webRequest.getParameter("page");
+                String sizeParam = webRequest.getParameter("size");
+
+                int page = DEFAULT_PAGE;
+                int size = DEFAULT_PAGE_SIZE;
+                try {
+                    if (pageParam != null) {
+                        if (!pageParam.matches("\\d+")) {
+                            throw new NumberFormatException(ErrorMessage.NEGATIVE_PAGE_VALUE_EXCEPTION);
+                        }
+                        page = Integer.parseInt(pageParam);
+                    }
+                    if (sizeParam != null) {
+                        if (!sizeParam.matches("\\d+")) {
+                            throw new NumberFormatException(ErrorMessage.NEGATIVE_SIZE_VALUE_EXCEPTION);
+                        }
+                        size = Integer.parseInt(sizeParam);
+                    }
+                } catch (NumberFormatException ex) {
+                    throw new IllegalArgumentException(ErrorMessage.INVALID_SIZE_VALUE_EXCEPTION);
```

we can throw this exception if an invalid page as well, not only an invalid size

---

I wonder if we really need to throw an exception here?
It is better to use default values

---

This is the core  of the task: if the size field is entered incorrectly, adjust the setting to the default value and return a response about the incorrectly entered field